src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md

```diff
@@ -38,42 +42,72 @@ For detailed information about these settings, how to make them permanent, and i
+
+</DescriptionList>
+
+## Monitoring the load generator
+
+The previous section described how hardware limits the load your machine can generate.
+Along with understanding conceptually, you also should monitor resources while the test runs&mdash;especially when running tests for the first time.
+
+
+The easiest way to monitor the load generator is to open multiple terminals on the machine: one to run k6, and others to monitor the CPU, memory, and network .
+We recommend the following tools:
 
-Simple tests will use ~1-5MB per VU. (1000VUs = 1-5GB).
-Tests that are using file uploads, or load large JS modules, can consume tens of megabytes per VU.
-Keep in mind that each VU has a copy of all JS modules your test uses.
-If you need to share memory between VUs, consider using [SharedArray](/javascript-api/k6-data/sharedarray/), or an external data store, such as [Redis](/javascript-api/k6-experimental/redis/).
+- CPU and memory: [htop](https://en.wikipedia.org/wiki/Htop) or [nmon](https://nmon.sourceforge.net/).
+- Network: [iftop](https://en.wikipedia.org/wiki/Iftop) or [nmon](https://nmon.sourceforge.net/).
+
+If you prefer graphical applications, use the system monitoring tool from your OS (e.g. GNOME System
+Monitor or Plasma System Monitor), or standalone tools like [SysMonTask](https://github.com/KrispyCamel4u/SysMonTask).
+
+Here's a screenshot of 3 terminal sessions showing k6, iftop and htop.
+![k6 iftop htop](./images/large-scale-testing-3-terminals.png)
 
-## General advice
+### Monitoring guidelines
 
-### Resilient error handling
+As the test runs, these are the indicators to monitor:
+
+- **CPU utilization stays within 80%.**
+  If all CPU cores are 100% utilized during the test run, you might notice performance degradation in your test results.
+
+- **Network utilization is at an acceptable level.**
+Depending on your test, you might expect to fully saturate the network bandwidth, or this might be a signal that your test is bound by the available bandwidth. In other scenarios, you might want to minimize network usage to keep costs down.
+
+- **Memory utilization doesn't exceed 90%.**
+If you're close to exhausting available physical RAM, the system might start swapping memory to disk, which will affect performance and system stability. In extreme cases, running out of memory on Linux will cause the system to end the k6 process.
+
+## Error handling should be resilient
 
 When running large stress tests, your script can't assume anything about the HTTP response.
-Often performance tests are written with a "happy path" in mind.
-For example, a "happy path" check like the one below is something that we often see in k6 scripts.
+It's a mistake to script a performance test with only the [happy path](/misc/glossary#happy-path "The default behavior that happens when the system returns no errors") in mind.
```

Comment:
I wouldn't phrase it as an outright "mistake". It's certainly an oversight, and it's good practice to think about failure scenarios.

Can we rephrase this in a similar way as it was before?